### ISSUE TITLE
Sorting on lists puts blanks last

### ISSUE THREAD

**Issue by sebgrebe** [rtcharity/eahub.org#124]
[Suggested](https://docs.google.com/spreadsheets/d/1qotdbleP9NHHieVsFkO2LQ6qeI43wBF2P0m_2NOe-xw/edit?ts=5c43311d#gid=0) by Marcin

**Comment by Richenda** [comment]
Great idea

**Comment by taymonbeal** [comment]
Is it okay for this to be postponed to Plato?

**Comment by taymonbeal** [comment]
Moving to Plato because there shouldn't be a lot of blank public profiles at launch anyway.

**Comment by sebgrebe** [comment]
moved to Plato, for reasons see [here](https://github.com/rtcharity/eahub.org/pull/633)

**Comment by sebgrebe** [comment]
Originally, we wanted to put blanks last. As discussed [here](https://github.com/rtcharity/eahub.org/pull/633#pullrequestreview-228671689), the idea was to do this on the database through order_by. While I found a way to do this, the problem is that this ordering is overrident client-side by the plugin that renders the profiles and groups table (dataTable).

There is a way to override the dataTable ordering such that blanks are put last (see [here](https://datatables.net/forums/discussion/54843/primary-secondary-sort-null-first-if-column-is-desc-or-null-last-if-column-is-asc)). However, in our specific context an implementation of this would be brittle (due to the fact that the fields were ordering are not just names, but actually contain links and other HTML). I decided that it would not be a good idea to implement this as it's too risky: There is probably no benefit to this feature since there are probably no public profiles with blank names (explained [here](https://github.com/rtcharity/eahub.org/pull/633#issuecomment-484887722)), but there is a risk that due to the brittleness of the code it would mess with the ordering.

Hence, I suggest to simply filter out profiles that have blank names so that they're not listed in the first place. We shouldn't have any public profiles with blank names anyway, and if we do, it seems best not to show them on the hub.
